oscpy/parser.py

```diff
@@ -145,6 +161,8 @@ def format_message(address, values, encoding='', encoding_errors='strict'):
                 if b'%i' in f:
                     v += b'\0'
                     f = f % padded(len(v), padsizes[cls])
+                if b'm' in tag:
+                    values[i] = format_midi(v)
```

Comment:
I think i avoided mutating the data from the message in case the user uses it for other things than just passing to oscpy, that's why we copy it if we need to encode it. keeping that will certainly make the code a bit less nice, unless we decide to go for copying regardless of necessity, which would certainly create a performance penalty, not sure what's the best course of action there.

Reply:
Absolutely right. Additional extended types (like the rgba color) will also require special casing. The problem is that the encoding can be set as a parameter, but tags are not known before the "parsing loop".

Reply:
pussed the full change on feature/format_midi_improve_format_message because a few more changes were needed, but the full test suite passes.